### PR TITLE
Chapter 5

### PR DESCRIPTION
**Pull request by tungcd**
Finish layout and routes

### CODE REVIEW

**Comment by tungcd** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -35,6 +35,7 @@ gem 'jquery-rails', '4.3.1'
 
 # Reduces boot times through caching; required in config/boot.rb
```
Comment (huuphuc93) [inline Gemfile]: remove all comments in this file

File: Gemfile [hunk 2]
```diff
@@ -35,6 +35,7 @@ gem 'jquery-rails', '4.3.1'
...
 gem 'bootsnap', '>= 1.1.0', require: false
```
Comment (huuphuc93) [inline Gemfile]: use " instead of '

File: app/assets/javascripts/users.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (huuphuc93) [inline app/assets/javascripts/users.coffee]: remove this file

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -0,0 +1,98 @@
+@import "bootstrap-sprockets";
+@import "bootstrap";
+
+/* mixins, variables, etc. */
+
+$gray-medium-light: #eaeaea;
```
Comment (huuphuc93) [inline app/assets/stylesheets/custom.scss]: move color variable to external file like color.scss and import to this file

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  float: left;
+  margin-right: 10px;
+  font-size: 1.7em;
+  color: white;
+  text-transform: uppercase;
+  letter-spacing: -1px;
+  padding-top: 9px;
+  font-weight: bold;
+  &:hover {
+    color: white;
```
Comment (huuphuc93) [inline app/assets/stylesheets/custom.scss]: use color variable

File: app/assets/stylesheets/users.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (huuphuc93) [inline app/assets/stylesheets/users.scss]: remove this file

File: app/views/layouts/_footer.html.erb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    by <a href="http://www.michaelhartl.com/">Michael Hartl</a>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to "About",   about_path %></li>
+      <li><%= link_to "Contact", contact_path %></li>
+      <li><a href="http://news.railstutorial.org/">News</a></li>
+    </ul>
+  </nav>
+</footer>
```
Comment (huuphuc93) [inline app/views/layouts/_footer.html.erb]: add the break line at the end of this file

File: app/views/layouts/_footer.html.erb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    The <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
```
Comment (huuphuc93) [inline app/views/layouts/_footer.html.erb]: Use gem i18n, move all text to file en.yml

File: app/views/layouts/_header.html.erb [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+    <%= link_to "sample app", root_path, id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home",    root_path %></li>
+        <li><%= link_to "Help",    help_path %></li>
+        <li><%= link_to "Log in", '#' %></li>
+      </ul>
+    </nav>
+  </div>
+</header>
```
Comment (huuphuc93) [inline app/views/layouts/_header.html.erb]: add the break line at the end of this file

File: app/views/layouts/_header.html.erb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "sample app", root_path, id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home",    root_path %></li>
```
Comment (huuphuc93) [inline app/views/layouts/_header.html.erb]: use a spaces between "Home", and root_path

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -6,8 +6,13 @@
     <%= stylesheet_link_tag    'application', media: 'all',
```
Comment (huuphuc93) [inline app/views/layouts/application.html.erb]: use " instead of ' in this whole file

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -6,8 +6,13 @@
...
   <body>
-    <%= yield %>
+    <%= render 'layouts/header' %>
+    <div class="container">
+      <%= yield %>
+      <%= render 'layouts/footer' %>
+    </div>
   </body>
-</html>
+</html>
```
Comment (huuphuc93) [inline app/views/layouts/application.html.erb]: add the break line at the end of this file

File: app/views/static_pages/contact.html.erb [hunk 13]
```diff
@@ -0,0 +1,6 @@
+<% provide(:title, 'Contact') %>
```
Comment (huuphuc93) [inline app/views/static_pages/contact.html.erb]: remove (), use " instead of '

File: app/views/static_pages/contact.html.erb [hunk 14]
```diff
@@ -0,0 +1,6 @@
...
+<h1>Contact</h1>
+<p>
+  Contact the Ruby on Rails Tutorial about the sample app at the
+  <a href="http://www.railstutorial.org/contact">contact page</a>.
+</p>
```
Comment (huuphuc93) [inline app/views/static_pages/contact.html.erb]: add the break line at the end of this file

File: app/views/static_pages/contact.html.erb [hunk 15]
```diff
@@ -0,0 +1,6 @@
+<% provide(:title, 'Contact') %>
+<h1>Contact</h1>
+<p>
+  Contact the Ruby on Rails Tutorial about the sample app at the
```
Comment (huuphuc93) [inline app/views/static_pages/contact.html.erb]: I18n

File: app/views/static_pages/home.html.erb [hunk 16]
```diff
@@ -1,7 +1,14 @@
...
+    This is the home page for the
+    <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
+    sample application.
+  </h2>
+
+  <%= link_to "Sign up now!", signup_path, class: "btn btn-lg btn-primary" %>
+</div>
+
+<%= link_to image_tag("rails.png", alt: "Rails logo"),
+            'http://rubyonrails.org/' %>
```
Comment (huuphuc93) [inline app/views/static_pages/home.html.erb]: add the break line at the end of this file

File: app/views/static_pages/home.html.erb [hunk 17]
```diff
@@ -1,7 +1,14 @@
-<% provide(:title, "Home") %>
-<h1>Sample App</h1>
-<p>
-  This is the home page for the
-  <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
-  sample application.
-</p>
\ No newline at end of file
+<div class="center jumbotron">
+  <h1>Welcome to the Sample App</h1>
```
Comment (huuphuc93) [inline app/views/static_pages/home.html.erb]: I18n

File: app/views/users/new.html.erb [hunk 18]
```diff
@@ -0,0 +1,3 @@
+<% provide(:title, 'Sign up') %>
```
Comment (huuphuc93) [inline app/views/users/new.html.erb]: remove (), use " instead of '

File: app/views/users/new.html.erb [hunk 19]
```diff
@@ -0,0 +1,3 @@
...
+<h1>Sign up</h1>
+<p>This will be a signup page for new users.</p>
```
Comment (huuphuc93) [inline app/views/users/new.html.erb]: add the break line at the end of this file

File: app/views/users/new.html.erb [hunk 20]
```diff
@@ -0,0 +1,3 @@
+<% provide(:title, 'Sign up') %>
+<h1>Sign up</h1>
```
Comment (huuphuc93) [inline app/views/users/new.html.erb]: I18n

File: config/routes.rb [hunk 21]
```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
+  get 'users/new'
```
Comment (huuphuc93) [inline config/routes.rb]: use " instead of '

File: config/routes.rb [hunk 22]
```diff
@@ -1,6 +1,8 @@
...
   root 'static_pages#home'
-  get  'static_pages/home'
-  get  'static_pages/help'
-  get  'static_pages/about'
-end
+  get  '/help',    to: 'static_pages#help'
+  get  '/about',   to: 'static_pages#about'
+  get  '/contact', to: 'static_pages#contact'
+  get  '/signup',  to: 'users#new'
+end
```
Comment (huuphuc93) [inline config/routes.rb]: add the break line at the end of this file

File: config/routes.rb [hunk 23]
```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
+  get 'users/new'
   root 'static_pages#home'
-  get  'static_pages/home'
-  get  'static_pages/help'
-  get  'static_pages/about'
-end
+  get  '/help',    to: 'static_pages#help'
```
Comment (huuphuc93) [inline config/routes.rb]: use a space between '/help', and to

File: test/controllers/users_controller_test.rb [hunk 24]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
```
Comment (huuphuc93) [inline test/controllers/users_controller_test.rb]: move folder test to gitignore, remove this file

**Comment by huuphuc93** [comment]
commented

**Comment by tungcd** [comment]
ready

File: app/assets/stylesheets/color.scss [hunk 25]
```diff
@@ -0,0 +1 @@
+$gray-medium-light: #eaeaea;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/color.scss]: Add a line break at the end of a file

File: app/assets/stylesheets/custom.scss [hunk 26]
```diff
@@ -0,0 +1,94 @@
+@import "bootstrap-sprockets";
+@import "bootstrap";
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Use single quote in CSS

File: app/assets/stylesheets/custom.scss [hunk 27]
```diff
@@ -0,0 +1,94 @@
...
+  line-height: 1.7em;
+}
+
+/* header */
+
+#logo {
+  float: left;
+  margin-right: 10px;
+  font-size: 1.7em;
+  color: white;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Use color variable

File: app/assets/stylesheets/custom.scss [hunk 28]
```diff
@@ -0,0 +1,94 @@
...
+  }
+  ul {
+    float: right;
+    list-style: none;
+    li {
+      float: left;
+      margin-left: 15px;
+    }
+  }
+}
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Add a line break at the end of a file

File: app/controllers/static_pages_controller.rb [hunk 29]
```diff
@@ -7,5 +7,8 @@ def help
 
    def about
   end
+
+  def contact
+  end
```
Comment (mdmahmudurrahman) [inline app/controllers/static_pages_controller.rb]: If an action don't have a body
`def contact; end `

File: app/controllers/users_controller.rb [hunk 30]
```diff
@@ -0,0 +1,4 @@
+class UsersController < ApplicationController
+  def new
+  end
```
Comment (mdmahmudurrahman) [inline app/controllers/users_controller.rb]: If an action don't have a body
`def contact; end`

File: app/views/layouts/_footer.html.erb [hunk 31]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    <%= t(".the")%> <a href="http://www.railstutorial.org/"><%= t(".ruby")%></a>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: Remove optional ()

File: app/views/layouts/_footer.html.erb [hunk 32]
```diff
@@ -0,0 +1,13 @@
...
+    <%= t(".by")%> <a href="http://www.michaelhartl.com/"><%= t(".name")%></a>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: Same here

File: app/views/layouts/_footer.html.erb [hunk 33]
```diff
@@ -0,0 +1,13 @@
...
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to t(".about"), about_path %></li>
+      <li><%= link_to t(".contact"), contact_path %></li>
+      <li><a href="http://news.railstutorial.org/"><%= t(".news")%></a></li>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: Same here

File: app/views/layouts/_header.html.erb [hunk 34]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to t(".sampleapp"), root_path, id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to t(".home"), root_path %></li>
+        <li><%= link_to t(".help"), help_path %></li>
+        <li><%= link_to t(".login"), '#' %></li>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: Use double quote in ruby

File: app/views/layouts/application.html.erb [hunk 35]
```diff
@@ -3,11 +3,15 @@
   <head>
     <title><%= full_title(yield(:title)) %></title>
     <%= csrf_meta_tags %>
-    <%= stylesheet_link_tag    'application', media: 'all',
-                                              'data-turbolinks-track': 'reload' %>
-    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <%= stylesheet_link_tag    "application", media: "all",
+                                              "data-turbolinks-track": "reload" %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/application.html.erb]: Indentation

File: app/views/static_pages/about.html.erb [hunk 36]
```diff
@@ -1,11 +1,10 @@
 <% provide(:title, "About") %>
 <h1>About</h1>
 <p>
-  The <a href="http://www.railstutorial.org/"><em>Ruby on Rails
-  Tutorial</em></a> is a
+  The <a href="http://www.railstutorial.org/"><em>Ruby on RailsTutorial</em></a> is a
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: I18n

File: app/views/static_pages/about.html.erb [hunk 37]
```diff
@@ -1,11 +1,10 @@
 <% provide(:title, "About") %>
 <h1>About</h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: I18n

File: app/views/static_pages/about.html.erb [hunk 38]
```diff
@@ -1,11 +1,10 @@
 <% provide(:title, "About") %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: Implement I18n

File: app/views/static_pages/contact.html.erb [hunk 39]
```diff
@@ -0,0 +1,6 @@
+<% provide :title, "Contact" %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/contact.html.erb]: I18n

File: app/views/static_pages/contact.html.erb [hunk 40]
```diff
@@ -0,0 +1,6 @@
...
+<h1><%= t(".contact")%></h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/contact.html.erb]: Remove optional ()

File: app/views/static_pages/contact.html.erb [hunk 41]
```diff
@@ -0,0 +1,6 @@
...
+<p>
+  <%= t(".contactattheruby")%>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/contact.html.erb]: Same here

File: app/views/static_pages/contact.html.erb [hunk 42]
```diff
@@ -0,0 +1,6 @@
...
+  <a href="http://www.railstutorial.org/contact"><%= t(".contactpage")%></a>.
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/contact.html.erb]: Same here

File: app/views/static_pages/home.html.erb [hunk 43]
```diff
@@ -1,7 +1,11 @@
-<% provide(:title, "Home") %>
-<h1>Sample App</h1>
-<p>
-  This is the home page for the
-  <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
-  sample application.
-</p>
\ No newline at end of file
+<div class="center jumbotron">
+  <h1><%= t(".welcome")%></h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Same here

File: app/views/static_pages/home.html.erb [hunk 44]
```diff
@@ -1,7 +1,11 @@
...
+  <h2>
+    <%= t(".this")%>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Same here

File: app/views/static_pages/home.html.erb [hunk 45]
```diff
@@ -1,7 +1,11 @@
...
+    <a href="http://www.railstutorial.org/"><%= t(".ruby")%></a>
+    <%= t(".sample")%>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Same here

File: app/views/static_pages/home.html.erb [hunk 46]
```diff
@@ -1,7 +1,11 @@
...
+  </h2>
+  <%= link_to t(".signupnow"), signup_path, class: "btn btn-lg btn-primary" %>
+</div>
+<%= link_to image_tag("rails.png", alt: "Rails logo"),
+            'http://rubyonrails.org/' %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: 1. Indentation
2. Use double quote

File: app/views/users/new.html.erb [hunk 47]
```diff
@@ -0,0 +1,3 @@
+<% provide :title, t(".signup") %>
+<h1><%= t(".signup")%></h1>
+<p><%= t(".this")%></p>
```
Comment (mdmahmudurrahman) [inline app/views/users/new.html.erb]: Please remove optional ()

File: config/routes.rb [hunk 48]
```diff
@@ -1,6 +1,8 @@
...
-  root 'static_pages#home'
-  get  'static_pages/home'
-  get  'static_pages/help'
-  get  'static_pages/about'
+  get "users/new"
+  root "static_pages#home"
+  get  "/help", to: "static_pages#help"
+  get  "/about", to: "static_pages#about"
+  get  "/contact", to: "static_pages#contact"
+  get  "/signup", to: "users#new"
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: 1 space between the parameters

File: config/routes.rb [hunk 49]
```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
-  root 'static_pages#home'
-  get  'static_pages/home'
-  get  'static_pages/help'
-  get  'static_pages/about'
+  get "users/new"
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Do you need this route? If so, then why?

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by tungcd** [comment]
ready

File: app/helpers/application_helper.rb [hunk 50]
```diff
@@ -1,6 +1,5 @@
 module ApplicationHelper
 
-  # Returns the full title on a per-page basis.
   def full_title(page_title = '')
```
Comment (huuphuc93) [inline app/helpers/application_helper.rb]: remove ()

File: app/helpers/application_helper.rb [hunk 51]
```diff
@@ -1,6 +1,5 @@
...
     base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (huuphuc93) [inline app/helpers/application_helper.rb]: use gem i18n move this text to en.yml

File: app/views/layouts/_footer.html.erb [hunk 52]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    <%= t ".the"%> <a href="http://www.railstutorial.org/"><%= t ".ruby"%></a>
```
Comment (huuphuc93) [inline app/views/layouts/_footer.html.erb]: use link_to instead of a tag

File: app/views/layouts/application.html.erb [hunk 53]
```diff
@@ -3,11 +3,14 @@
   <head>
     <title><%= full_title(yield(:title)) %></title>
```
Comment (huuphuc93) [inline app/views/layouts/application.html.erb]: full_title(yield(:title)) => full_title yield(:title)

File: app/views/static_pages/contact.html.erb [hunk 54]
```diff
@@ -0,0 +1,6 @@
+<% provide :title, t(".contact")  %>
+<h1><%= t  ".contact" %></h1>
```
Comment (huuphuc93) [inline app/views/static_pages/contact.html.erb]: 1 spaces instead of  2

File: config/routes.rb [hunk 55]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
   root 'static_pages#home'
-  get  'static_pages/home'
-  get  'static_pages/help'
-  get  'static_pages/about'
-end
+  get '/help', to: 'static_pages#help'
```
Comment (huuphuc93) [inline config/routes.rb]: use " instead of '

File: config/routes.rb [hunk 56]
```diff
@@ -1,6 +1,7 @@
...
+  get '/about', to: 'static_pages#about'
+  get '/contact', to: 'static_pages#contact'
+  get '/signup', to: 'users#new'
+end
```
Comment (huuphuc93) [inline config/routes.rb]: end of file

File: config/locales/en.yml [hunk 57]
```diff
@@ -1,33 +1,47 @@
-# Files in the config/locales directory are used for internationalization
```
Comment (huuphuc93) [inline config/locales/en.yml]: add file vi.yml to use vietnamese language

**Comment by huuphuc93** [comment]
commented

File: Gemfile [hunk 58]
```diff
@@ -1,62 +1,36 @@
...
+gem "puma", "~> 3.11"
+gem "sass-rails", "~> 5.0"
+gem "uglifier", ">= 1.3.0"
+gem "coffee-rails", "~> 4.2"
+gem "turbolinks", "~> 5"
+gem "jbuilder", "~> 2.5"
+gem "jquery-rails", "4.3.1"
+gem "bootsnap", ">= 1.1.0", require: false
+gem "bootstrap-sass", "3.3.7"
+gem 'config'
```
Comment (huuphuc93) [inline Gemfile]: use " instead of '

File: app/helpers/application_helper.rb [hunk 59]
```diff
@@ -1,7 +1,6 @@
 module ApplicationHelper
 
-  # Returns the full title on a per-page basis.
-  def full_title(page_title = '')
+  def full_title page_title = ''
     base_title = "Ruby on Rails Tutorial Sample App"
```
Comment (huuphuc93) [inline app/helpers/application_helper.rb]: I18n

File: app/views/static_pages/home.html.erb [hunk 60]
```diff
@@ -1,7 +1,10 @@
...
+<div class="center jumbotron">
+  <h1><%= t ".welcome"%></h1>
+  <h2>
+    <%= t ".this"%>
+    <%= link_to t(".ruby"), Settings.railstutorial %>
+    <%= t ".sample"%>
+  </h2>
+  <%= link_to t(".sign_up_now"), "#", class: "btn btn-lg btn-primary" %>
+</div>
+<%= link_to image_tag("rails.png", alt: "Rails logo"),"http://rubyonrails.org/" %>
```
Comment (huuphuc93) [inline app/views/static_pages/home.html.erb]: move "http://rubyonrails.org/" to Settings

File: app/views/users/new.html.erb [hunk 61]
```diff
@@ -0,0 +1,3 @@
+<% provide :title, t(".signup") %>
+<h1><%= t ".signup"%></h1>
```
Comment (huuphuc93) [inline app/views/users/new.html.erb]: t ".signup"%> => t ".signup" %>

File: config/application.rb [hunk 62]
```diff
@@ -10,6 +10,7 @@ module AppSample
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
```
Comment (huuphuc93) [inline config/application.rb]: remove all comments in this file

File: config/settings.yml [hunk 63]
```diff
@@ -0,0 +1,9 @@
+rubyonrails: "http://rubyonrails.org/"
+screencasts: "http://screencasts.railstutorial.org/"
+railsbook: "http://www.railstutorial.org/book"
+railstutorial: "http://www.railstutorial.org/"
+rails_contact: "http://www.railstutorial.org/contact"
+rails_help: "http://www.railstutorial.org/help"
+rails_book: "http://www.railstutorial.org/book"
+michael: "http://www.michaelhartl.com/"
+news: "http://news.railstutorial.org/"
```
Comment (huuphuc93) [inline config/settings.yml]: end of file

**Comment by huuphuc93** [comment]
commented
comment nhỏ

File: app/views/static_pages/about.html.erb [hunk 64]
```diff
@@ -1,11 +1,10 @@
...
-  to teach web development with
-  <a href="http://rubyonrails.org/">Ruby on Rails</a>.
-  This is the sample application for the tutorial.
-</p>
\ No newline at end of file
+  <%= link_to t(".rubys"), Settings.railstutorial %> <%= t".is_a" %>
+  <%= link_to t(".book"), Settings.railsbook %> <%= t".and" %>
+  <%= link_to t(".screencasts"), Settings.screencasts %>
+  <%= t ".to" %>
+  <%= link_to t(".ruby"), Settings.rubyonrails  %>.
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: maximum 1 space before the closing erb tag

File: app/views/static_pages/home.html.erb [hunk 65]
```diff
@@ -1,7 +1,10 @@
-<% provide(:title, "Home") %>
-<h1>Sample App</h1>
-<p>
-  This is the home page for the
-  <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
-  sample application.
-</p>
\ No newline at end of file
+<div class="center jumbotron">
+  <h1><%= t".welcome"%></h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Add a white space before closing the erb tag

File: app/views/static_pages/home.html.erb [hunk 66]
```diff
@@ -1,7 +1,10 @@
...
+  <h2>
+    <%= t".this"%>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Same here

File: app/views/static_pages/home.html.erb [hunk 67]
```diff
@@ -1,7 +1,10 @@
...
+    <%= link_to t(".ruby"), Settings.rubyonrails %>
+    <%= t".sample"%>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Same here

File: app/views/users/new.html.erb [hunk 68]
```diff
@@ -0,0 +1,3 @@
+<% provide :title, t(".signup") %>
+<h1><%= t".signup"%></h1>
+<p><%= t".this"%></p>
```
Comment (mdmahmudurrahman) [inline app/views/users/new.html.erb]: Same here

File: app/views/users/new.html.erb [hunk 69]
```diff
@@ -0,0 +1,3 @@
+<% provide :title, t(".signup") %>
+<h1><%= t".signup"%></h1>
```
Comment (mdmahmudurrahman) [inline app/views/users/new.html.erb]: Same here

File: config/locales/en.yml [hunk 70]
```diff
@@ -1,33 +1,53 @@
...
+      contact: "Contact"
+      contact_at_the_ruby: "Contact the Ruby on Rails Tutorial about the sample app at the"
+      contact_page: "contact page"
+  users:
+    new:
+      sign_up: "Sign up"
+      this: "This will be a signup page for new users."
+
+      
+
```
Comment (mdmahmudurrahman) [inline config/locales/en.yml]: Please remove last 3 empty lines

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by tungcd** [comment]
ready

File: app/views/static_pages/about.html.erb [hunk 71]
```diff
@@ -1,11 +1,10 @@
...
-  <a href="http://rubyonrails.org/">Ruby on Rails</a>.
-  This is the sample application for the tutorial.
-</p>
\ No newline at end of file
+  <%= link_to t(".rubys"), Settings.railstutorial %> <%= t".is_a" %>
+  <%= link_to t(".book"), Settings.railsbook %> <%= t".and" %>
+  <%= link_to t(".screencasts"), Settings.screencasts %>
+  <%= t ".to" %>
+  <%= link_to t(".ruby"), Settings.rubyonrails %>.
+  <%= t".this" %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: `<%= t ".this" %>`

File: app/views/static_pages/about.html.erb [hunk 72]
```diff
@@ -1,11 +1,10 @@
...
-  Tutorial</em></a> is a
-  <a href="http://www.railstutorial.org/book">book</a> and
-  <a href="http://screencasts.railstutorial.org/">screencast series</a>
-  to teach web development with
-  <a href="http://rubyonrails.org/">Ruby on Rails</a>.
-  This is the sample application for the tutorial.
-</p>
\ No newline at end of file
+  <%= link_to t(".rubys"), Settings.railstutorial %> <%= t".is_a" %>
+  <%= link_to t(".book"), Settings.railsbook %> <%= t".and" %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: `<%= t ".and" %>`

File: app/views/static_pages/about.html.erb [hunk 73]
```diff
@@ -1,11 +1,10 @@
-<% provide(:title, "About") %>
-<h1>About</h1>
+<% provide :title, t(".about") %>
+<h1><%= t".about" %></h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: `<%= t ".about" %>`

File: app/views/static_pages/contact.html.erb [hunk 74]
```diff
@@ -0,0 +1,6 @@
+<% provide :title, t(".contact") %>
+<h1><%= t".contact" %></h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/contact.html.erb]: `<%= t ".contact" %>`

File: app/views/static_pages/home.html.erb [hunk 75]
```diff
@@ -1,7 +1,10 @@
-<% provide(:title, "Home") %>
-<h1>Sample App</h1>
-<p>
-  This is the home page for the
-  <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
-  sample application.
-</p>
\ No newline at end of file
+<div class="center jumbotron" >
+  <h1><%= t".welcome" %></h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: `<%= t ".welcome" %>`

File: app/views/static_pages/home.html.erb [hunk 76]
```diff
@@ -1,7 +1,10 @@
...
+  <h2>
+    <%= t".this" %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: `<%= t ".this" %>`

File: app/views/static_pages/home.html.erb [hunk 77]
```diff
@@ -1,7 +1,10 @@
...
+    <%= link_to t(".ruby"), Settings.rubyonrails %>
+    <%= t".sample" %>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: `    <%= t ".sample" %>`

File: app/views/users/new.html.erb [hunk 78]
```diff
@@ -0,0 +1,3 @@
+<% provide :title, t(".signup") %>
+<h1><%= t".signup" %></h1>
+<p><%= t".this" %></p>
```
Comment (mdmahmudurrahman) [inline app/views/users/new.html.erb]: Same here

File: app/views/users/new.html.erb [hunk 79]
```diff
@@ -0,0 +1,3 @@
+<% provide :title, t(".signup") %>
+<h1><%= t".signup" %></h1>
```
Comment (mdmahmudurrahman) [inline app/views/users/new.html.erb]: Same here

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by tungcd** [comment]
ready

File: app/views/layouts/_footer.html.erb [hunk 80]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    <%= t ".the"%> <%= link_to t(".ruby"), Settings.railstutorial %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: Add a space before closing an erb tag
`<%= t ".the" %>`

File: app/views/layouts/_footer.html.erb [hunk 81]
```diff
@@ -0,0 +1,13 @@
...
+    <%= t ".by"%> <%= link_to t(".name"), Settings.michael %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: The same here

File: app/views/static_pages/about.html.erb [hunk 82]
```diff
@@ -1,11 +1,10 @@
-<% provide(:title, "About") %>
-<h1>About</h1>
+<% provide :title, t(".about")  %>
+<h1><%= t ".about"%></h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: The same here
`<%= t ".about" %>`

File: app/views/static_pages/about.html.erb [hunk 83]
```diff
@@ -1,11 +1,10 @@
...
-  to teach web development with
-  <a href="http://rubyonrails.org/">Ruby on Rails</a>.
-  This is the sample application for the tutorial.
-</p>
\ No newline at end of file
+  <%= link_to t(".rubys"), Settings.railstutorial %> <%= t ".is_a" %>
+  <%= link_to t(".book"), Settings.railsbook %> <%= t ".and" %>
+  <%= link_to t(".screencasts"), Settings.screencasts %>
+  <%= t ".to" %>
+  <%= link_to t(".ruby"), Settings.rubyonrails  %>.
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/about.html.erb]: Please remove redundant space before closing the erb tag, just 1 space is allowed

**Comment by mdmahmudurrahman** [comment]
commented

**Pull request merged**